### ISSUE TITLE
Updating k8s to etcd 3.0.3 causes 'use of internal package not allowed'

### ISSUE THREAD

**Issue by timothysc** [coreos/etcd#6027]
```
package k8s.io/kubernetes/cmd/hyperkube
	imports k8s.io/kubernetes/cmd/kube-apiserver/app
	imports k8s.io/kubernetes/pkg/genericapiserver
	imports k8s.io/kubernetes/pkg/storage/storagebackend/factory
	imports github.com/coreos/etcd/clientv3
	imports github.com/coreos/etcd/etcdserver/etcdserverpb
	imports github.com/gengo/grpc-gateway/runtime
	imports github.com/grpc-ecosystem/grpc-gateway/runtime/internal: use of internal package not allowed
!!! Error in /home/tsclair/work/spaces/go/src/k8s.io/kubernetes/hack/lib/golang.sh:504
  'go install "${goflags[@]:+${goflags[@]}}" -gcflags "${gogcflags}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 1
Call stack:
  1: /home/tsclair/work/spaces/go/src/k8s.io/kubernetes/hack/lib/golang.sh:504 kube::golang::build_binaries_for_platform(...)
  2: /home/tsclair/work/spaces/go/src/k8s.io/kubernetes/hack/lib/golang.sh:678 kube::golang::build_binaries(...)
  3: hack/make-rules/build.sh:26 main(...)
Exiting with status 1
```

**Comment by gyuho** [comment]
What go version does this build with?

**Comment by heyitsanthony** [comment]
@timothysc It looks like something is vendoring / importing gengo/grpc-gateway. The repo has moved to grpc-ecosystem/grpc-gateway and etcd makes no reference to the old repo path. The deps in k8s are probably stale.

**Comment by timothysc** [comment]
I built with go1.6.2 and I believe it is transitive from etcd deps that reference:
 
"github.com/gengo/grpc-gateway/runtime"
"github.com/gengo/grpc-gateway/utilities"

**Comment by heyitsanthony** [comment]
@timothysc OK, I see the vendor update happened in master slightly after 3.0.3. Should be fixed in 3.0.4.

**Comment by timothysc** [comment]
eta on 3.0.4?

**Comment by xiang90** [comment]
Next Wednesday.

**Comment by timothysc** [comment]
will it be based on master?  if so, I can just tread ahead for now.

**Comment by xiang90** [comment]
no. current master is not stable. that is why we need to back-port to branch 3.0. there is a few grpc related issues we need to figure out. if you do not need a release, we probably can backport the import fix now. /cc @gyuho

**Comment by gyuho** [comment]
@xiang90 The rename patch is already backported https://github.com/coreos/etcd/commit/be541f36416c884576c041a83f127e404413697e to `release-3.0` branch.

But we put off publishing that release to next Wednesday because of grpc performance issue.

**Comment by xiang90** [comment]
@timothysc OK. If you try the current 3.0 release branch head (not the 3.0.3 tag), it should work.

**Comment by timothysc** [comment]
little further: 

```
# k8s.io/kubernetes/vendor/github.com/grpc-ecosystem/grpc-gateway/runtime/internal
vendor/github.com/grpc-ecosystem/grpc-gateway/runtime/internal/stream_chunk.pb.go:29: undefined: proto.ProtoPackageIsVersion2
vendor/github.com/grpc-ecosystem/grpc-gateway/runtime/internal/stream_chunk.pb.go:49: undefined: proto.RegisterFile
```

**Comment by gyuho** [comment]
@timothysc Yeah `grpc-gateway` imports `github.com/golang/protobuf` (https://github.com/coreos/etcd/blob/master/cmd/Godeps/Godeps.json#L90). Does Kubernetes have `golang/protobuf` in its vendor? So now we need both `gogo/protobuf` and `golang/protobuf`.

**Comment by timothysc** [comment]
Ahh looks like transitive mismatch on the dep.. I'll resolve shortly.

**Comment by timothysc** [comment]
k, I've got it mostly figured out.  We should still leave this issue open until 3.0.4 is released and then I will update the deps.

**Comment by gyuho** [comment]
Just released etcd v3.0.4 at https://github.com/coreos/etcd/releases/tag/v3.0.4.

Thanks.

**Comment by hongchaodeng** [comment]
@timothysc 
I'm closing this and converge the discussion in https://github.com/coreos/rkt/issues/2971

Thanks @gyuho for the work!
